### DISCUSSION TITLE
Visreg

### DISCUSSION THREAD

**Discussion by Nathan-Hebert** [sdmTMB/sdmTMB#332] [Q&A] [answered]
What values does visreg use for the spatial/spatio-temporal random effects? Is there a way to specify the "time" variable to visreg?

Thanks!

**Comment by seananderson** [comment, marked as answer]
It would be turning them off / setting them to zero. I.e. setting `re_form = NA` in predict.sdmTMB.

https://github.com/pbs-assess/sdmTMB/blob/1dc2801aac4c9a5c1b848b50c52e4f36a9d28f6c/R/predict.R#L954-L968
(The internal code to make visreg work is kind of crazy.)

You could also consider now using `ggeffects::ggpredict()` instead.

https://github.com/pbs-assess/sdmTMB/blob/1dc2801aac4c9a5c1b848b50c52e4f36a9d28f6c/R/fit.R#L405

https://github.com/strengejacke/ggeffects/blob/master/tests/testthat/test-sdmTMB.R

There's a half-finished vignette that I should get out.

Or `ggeffects::ggeffect()`, assuming you aren't working with smoothers, which will be fast.
